### DISCUSSION TITLE
Vite react plugin not transforming process.env in library mode

### DISCUSSION THREAD

**Discussion by MaxWilson** [vitejs/vite-plugin-react#261] [Q&A] [answered]
I have an issue with vite and react. When I'm building against index.html everything works fine, but if I configure vite to build a JavaScript module then references to process.env.NODE_ENV get left in my code, causing exceptions at runtime. I know that the correct way is for the code to reference import.meta.env instead of process.env, and my working theory is that the vite-react-plugin normally takes care of this, but apparently not when building a library. (Is "library mode" the right thing to call it?)

```
  // config = { blahblahblah...,
  build: {
    manifest: true,
    outDir: 'publish',
    emptyOutDir: true,
    sourcemap: true,
    lib: {
      name: 'Arena',
      entry: 'dist/Main.jsx',
      formats: ['es'],
      fileName: 'Main'
    }
  },
```

Have you ever had this issue and how did you solve it?

**Comment by ArnaudBarre** [comment, marked as answer]
This is an expected behavior: https://github.com/vitejs/vite/pull/8090
See this comment if you want to continue using this kind of outdating pattern: https://github.com/vitejs/vite/pull/8090#issuecomment-1184929037
